expences-app-swiftui/Views/ExpenseEditingView.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  AddExpense.swift
+//  expences-app-swiftui
+//
+//  Created by Serhii Prykhodko on 16.10.2020.
+//
+
+import SwiftUI
+
+struct ExpenseEditingView: View {
+    
+    @Binding var expense: Expense?
+    @State var name = ""
+    @State var amount = ""
+    @State var showAlert = false
+    @Binding var personIndex: Int
+    @Binding var dayIndex: Int
+    @Binding var positionIndex: Int
+    @ObservedObject var expensesStore: ExpensesStore
+    var operation: OperationType
+    @State var navigationText = ""
+    @State var actionButtonText = ""
+    
+    var body: some View {
+        NavigationView {
+            Form {
+                Section(header: Text("Name")) {
+                    TextField("Enter name here", text: $name)
+                }
+                Section(header: Text("Amount")) {
+                    TextField("Enter amount here", text: $amount)
+                }
+            }
+            .onAppear(perform: {
+                checkTranferedExpense()
+                checkOperationType(operation: operation)
+            })
+            .navigationBarTitle(navigationText, displayMode: .inline)
+            .navigationBarItems(
+                leading:
+                    Button(action: { expense = nil }) {
+                        Text("Cancel")
+                    },
+                trailing:
+                    Button(action: { handleExpense(operation: operation) }) {
+                        Text(actionButtonText)
+                    }
+                    .disabled(self.name.isEmpty || self.amount.isEmpty)
+                    .alert(isPresented: $showAlert) {
+                        Alert(title: Text("Information"), message: Text("Entered amount is incorrect"), dismissButton: .cancel(Text("OK")))
+                    }
+            )
+        }
+    }
+    
+    private func handleExpense(operation: OperationType) {
+        let enteredAmount = checkEnteredValue(value: self.amount)
+        guard enteredAmount != nil else {
+            self.amount = ""
+            self.showAlert = true
+            
+            return
+        }
+        let expense = Expense(name: self.name, amount: enteredAmount!)
```

Comment:
you shouldn't have `enteredAmount!` here, please try `guard let enteredAmount = enteredAmount else {` above and then
```suggestion
        let expense = Expense(name: name, amount: enteredAmount)
```